utils/githooks/prepare-commit-msg.d/copy-commit-tags.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import subprocess
+from collections import OrderedDict
+
+# File generated by pre-commit hook.
+MSG_FILE='.amp-diff'
+
+class NotTag(Exception):
+    """Nothing"""
+    
+def get_tag_kv(line):
+    """Convert a line of test to a key/value"""
+
+    if ':' not in line:
+        raise NotTag
+    (raw_key, value) = line.split(':', maxsplit=1)
+    key = raw_key.strip()
+    if key in ('Date', 'Author', 'Signed-off-by', 'Merge', 'Co-authored-by'):
+        raise NotTag
+    if ' ' in key:
+        raise NotTag
+    return (key, value.strip())  
+
+def main():
+    input_file = sys.argv[1]
+
+    def add_text():
+        output = '# ------------------------ >8 ------------------------\n'
+        output += 'Skip-func-hw-test: true\n'
+        output += 'Skip-func-test: true\n'
+        output += 'Quick-Functional: true\n'
+        output += 'Test-tag: dfuse\n'
+        if os.path.exists(MSG_FILE):
+            with open(MSG_FILE, 'r') as new:
+                for newline in new.readlines():
+                    output += newline
+            os.unlink(MSG_FILE)
+        return output
+
+    if os.path.exists('.git/MERGE_HEAD'):
+        print('Not modifying merge-head')
+        return
+
+    tags = OrderedDict()
+    rc = subprocess.run(['git', 'log', '-1'], stdout=subprocess.PIPE)
```

Comment:
(pylint-subprocess-run-check) Using subprocess.run without explicitly set `check` is not recommended.